Security/Authentication/Token/PreAuthenticationJWTUserToken.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Lexik\Bundle\JWTAuthenticationBundle\Security\Authentication\Token;
+
+use Symfony\Component\Security\Guard\Token\PreAuthenticationGuardToken;
+
+/**
+ * PreAuthenticationJWTUserToken.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+final class PreAuthenticationJWTUserToken extends PreAuthenticationGuardToken
```

Comment:
I made it extend `PreAuthenticationGuardToken` primarily because [`JWTManager::decode()`](https://github.com/lexik/LexikJWTAuthenticationBundle/blob/guard_jwt_authenticator/Services/JWTManager.php#L83) that is used in [`getUser`](https://github.com/lexik/LexikJWTAuthenticationBundle/blob/guard_jwt_authenticator/Security/Guard/JWTTokenAuthenticator.php#L84) only accepts a`TokenInterface` instance, which makes sense to me, at least it doesn't worth to change it imho. Secondarily for benefiting of the `PreAuthenticationGuardToken::setAuthenticated()` method throwing an exception and the `$authenticated` property set to `false` in `__construct()`. 
This object inherit lots of properties and methods that aren't necessary, I just didn't find a lighter alternative for now :/